aws/src/main/java/org/apache/iceberg/aws/s3/S3OutputStream.java

```diff
@@ -328,6 +330,15 @@ private static InputStream uncheckedInputStream(File file) {
     }
   }
 
+  private void createStagingDirectoryIfNotExists() throws SecurityException {
+    if (!stagingDirectory.exists()) {
+      boolean createdStagingDirectory = stagingDirectory.mkdirs();
+      if (createdStagingDirectory) {
+        LOG.info("Successfully created staging directory: {}", stagingDirectory.getAbsolutePath());
+      }
```

Comment:
else, add a LOG saying "Staging directory {} creation failed, or it is created by another process"

Reply:
actually, we can do better than just logging. When creation fails, we can check directory existence again, if it exists then it's created by another process, otherwise it's still an issue, and we can throw an IOException with error message indicating staging directory creation fails for some unknown reasons.

Reply:
Agreed, I have made the changes accordingly.